pkg/kubefed2/federate/enable.go

```diff
@@ -124,32 +131,56 @@ func NewCmdFederateEnable(cmdOut io.Writer, config util.FedConfig) *cobra.Comman
 
 // Complete ensures that options are valid and marshals them if necessary.
 func (j *enableType) Complete(args []string) error {
+	j.federateDirective = &FederateDirective{}
```

Comment:
I think you wrote `NewFederateDirective()` to be used here, but missed using it.